auto_update.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/blang/semver"
+
+	"github.com/inconshreveable/go-update"
+)
+
+// RunUpdater checks if an update is due, checks if current version is outdated and performs update if needed
+func RunUpdater(app *TGFApplication) bool {
+	const autoUpdateFile = "tgfautoupdate"
+	if lastRefresh(autoUpdateFile) < 2*time.Hour {
```

Comment:
If we specify --update, the update must be done even if the last check occurred within the last two hours.

```suggestion
	if !app.AutoUpdate && lastRefresh(autoUpdateFile) < 2*time.Hour {
```